breathecode/events/tasks.py

```diff
@@ -199,3 +200,54 @@ def fix_live_class_dates(timeslot_id: int):
 
         starting_at += delta
         ending_at += delta
+
+
+@shared_task(bind=True, priority=TaskPriority.ACADEMY.value)
+def generate_event_recap(self, event_id: int):
+    """
+    Generate a recap of the event using rigobot AI.
+    This task will be triggered when an event changes to FINISHED status.
+    """
+    import re
+
+    event = Event.objects.filter(id=event_id).first()
+    if not event:
+        logger.error(f"Event {event_id} not found")
+        return
+
+    if event.recap:
+        logger.info(f"Event {event_id} already has a recap, skipping")
+        return
+
+    try:
+        with Service("rigobot", event_id) as s:
+            event_data = {
+                "event_title": event.title,
+                "event_description": event.description,
+                "event_type": event.tags[0] if event.tags and len(event.tags) > 0 else "",
+            }
+
+            response = s.post(
+                "/v1/prompting/completion/linked/event-recap/", json={"inputs": event_data, "execute_async": False}
+            )
+
+            if response.status_code >= 200 and response.status_code < 400:
+                result = response.json()
+                answer = result.get("answer", "")
+
+                match = re.search(r"<event-description>(.*?)</event-description>", answer, re.DOTALL)
+                recap_text = match.group(1).strip() if match else ""
+
+                if recap_text:
+                    try:
+                        Event.objects.filter(id=event_id).update(recap=recap_text)
+                    except Exception as e:
+                        logger.error(f"Error saving event {event_id} after getting recap: {e}", exc_info=True)
+                else:
+                    logger.warning(f"Recap for event {event_id} could not be extracted from answer: {answer[:100]}...")
+
+                logger.info(f"API call successful for event {event_id}, response: {response.text}")
+            else:
+                logger.error(f"Failed to generate recap: {response.status_code} - {response.text}")
```

Comment:
Right now we're not have anything like a `TaskError`, you could use `AbortTask` with it